src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -545,6 +546,17 @@ public ResponseEntity<SnapshotAccessRequestResponse> createSnapshotRequest(
             id, snapshotAccessRequest, userRequest.getEmail()));
   }
 
+  @Override
+  public ResponseEntity<EnumerateSnapshotAccessRequest> enumerateSnapshotRequests(UUID id) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userRequest,
+        IamResourceType.DATASET,
+        id.toString(),
+        IamAction.UPDATE_SNAPSHOT_BUILDER_SETTINGS);
```

Comment:
It seems like the `IamAction.UPDATE_SNAPSHOT_BUILDER_SETTINGS` may be getting overloaded in these access checks.  That might be okay for the time being, but I could see wanting more granular access control down the line (e.g. a user may be able to enumerate or fetch snapshot access requests without needing to have the heightened permissions require to modify snapshot builder settings).

Reply:
Yeah this is very true